grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1890,20 +1889,14 @@ class StockMovementService {
                 }
 
                 if (!requisitionItem) {
-                    throw new IllegalArgumentException("Could not find stock movement item with ID ${stockMovementItem.id}")
+                    throw new IllegalArgumentException("Could not find requisition item for stock movement item with ID ${stockMovementItem.id}")
                 }
 
-                if (requisitionItem.shipmentItems || requisitionItem.picklistItems) {
-                    removeShipmentAndPicklistItemsForModifiedRequisitionItem(requisitionItem)
-                }
-
-                log.info "Item revised " + requisitionItem.id
-
-                // Cannot cancel quantity if it has already been canceled
-                if (requisitionItem.quantityCanceled) {
-                    requisitionItem.undoChanges()
```

Comment:
I can't think of under what circumstances this would occur, but there might be cases where we aren't canceling the quantity so we don't want to undo the changes. But I'll defer to @awalkowiak because this code has changed a lot since it was first written.

Reply:
@jmiranda I tried to be as nit picky in the review as I could, I pulled it locally too, and it seems to be fine. The `undo` button basically under the hood is doing the:
```
removeShipmentAndPicklistItemsForModifiedRequisitionItem(requisitionItem)
requisitionItem.undoChanges()
```
so we are simply doing the `revert item` first here. I think Katarzyna will find all the edge cases with it (for me it looks good).

Btw @jmiranda I found NPE in your recent changes in the `requisitionItem.undoChanges()` here: https://github.com/openboxes/openboxes/blob/develop/grails-app/domain/org/pih/warehouse/requisition/RequisitionItem.groovy#L224. I am assuming, that it should be `substitutionItem.disableRefresh = Boolean.TRUE`, I'll push a fixing commit within this PR.

cc @mdpearson